src/allianceutils/util/camel_case.py

```diff
@@ -99,39 +118,49 @@ def process_path(parts: Sequence[str], candidates: Sequence[Dict]):
             part = parts[0]
             assert part != ''
 
-            new_candidates = []
+            new_candidates: list[IgnoreDict] = []
 
             for candidate in candidates:
+                assert isinstance(candidate, dict)  # TODO: remove
```

Comment:
is this to be removed now?

Reply:
Yes; that was a genuine "this is not fully typed and I just refactored it, are my assumptions correct?"

All the test cases are passing now so have removed it.